vignettes/publishing-extensions.Rmd

```diff
@@ -0,0 +1,83 @@
+  - [`TableauExtension.Enabled`](https://docs.rstudio.com/connect/admin/appendix/configuration/#TableauExtension) must be `True`.
+  - [`Server.Address`](https://docs.rstudio.com/connect/admin/appendix/configuration/#Server.Address) must be configured.
+
+See the [RStudio Connect Admin Guide](https://docs.rstudio.com/connect/admin/appendix/configuration/) for more information about configuring Connect.
+
+Extensions published to an incompatible RStudio Connect server will an error message that detects which of the above criteria are not met.
+
+![](files/warnings.png){width="75%"}
+
+### Access, Permissions, and Security {#access}
+
+Your RStudio Connect server must be [configured as an Analytics Extension in Tableau](#tableau). Tableau requires an [API key](https://docs.rstudio.com/connect/user/api-keys/) to authenticate with RStudio Connect. This API key will be used for all requests from Tableau to RStudio Connect, and the Tableau instance will have access to any content on Connect that the API can access.
+
+We recommend creating a "service account" (an account not used only by Tableau, not by any individual) on RStudio Connect, and configuring Tableau with API key for this account. If you name the service account "Tableau", simply add the "Tableau" to the [Access List](https://docs.rstudio.com/connect/user/content-settings/#content-access) for any content it should be able to access. This way, you can more finely control what content Tableau can access without affecting any users.
+
+## Configuring RStudio Connect as an Extension Server in Tableau #{tableau}
+
+Before you can use extensions hosted on an RStudio Connect server from Tableau, you'll need to register that server in Tableau's Extensions settings. you'll need to set up  The steps to set Tableau differ slightly depending on whether you're using Tableau Desktop or Tableau and Online.
+
+### Tableau Server / Tableau Online
+  1. Using an administrative account, login to Tableau Server/Online
+  2. Navigate to Settings, then Extensions
+  3. Under the heading 'Analytics Extensions', select 'Enable analytics extension for site'
+  4. Create a new connection and select the connection type of 'Analytics Extensions API'
+  5. Select whether you want to use SSL and enter the server Host and Port for your RStudio Connect server
+  6. Select 'Sign in with a username and password'. The username is 'rstudio-connect' and the password is any valid API key from RStudio Connect
+  8. Create / Save changes
+
+### Tableau Desktop
+  1. Navigate to Help, Settings and Performance, Manage Analytics Extension Connection...
+  2. Select 'TabPy/External API'
+  3. Set Server and Port to the address and port of the server running the API
+  4. If desired, select 'Sign in with a username and password'. The username is 'rstudio-connect' and the password is any valid API key from RStudio Connect
+  5. Select whether to Require SSL
+  6. Save changes
```

Comment:
We'll need to figure out how we handle changes here on the Tableau side. For example, in the most recent release of Tableau desktop, the setup process for analytics extensions is much more similar to the setup for Tableau Server/Online.

Reply:
Yeah, I figure this'll have to change over time. I just copied it pretty much verbatim from the in-extension setup.